### DISCUSSION TITLE
Need help for Checkmarx.Api cake plugin

### DISCUSSION THREAD

**Discussion by raysaik** [cake-build/cake#3752] [Q&A] [answered]
I am trying to incorporate "Checkmarx" Static code scans as a stage into my devops pipeline. Currently our code uses "cake" files to excute the stages (invoked by PowerShell).

I was checking the cake support for Checkmarx.Api but could not find any neither in the Checkmarx site or in the Cake website. The NuGet gallery has a tab for the cake addin - https://www.nuget.org/packages/Checkmarx.API/ 

but does not share any information on the contracts exposed.

So reaching out to the community to see if anyone has done any work on this or has any references. Any other way you have incorporated "Checkmarx" into your build pipeline (without directly using the plugin rather using the CxCLi) would also be helpful as well.

**Comment by nils-a** [comment, marked as answer]
@raysaik Cake scripts based on "normal" C#, so whatever the usage of Checkmarx.API, you can simply incorporate that in your cake scripts. Probably something like:

```cakescript
Task("Scan")
 .Does(() => 
{
  // place your code here..
});
```

As for using Checkmarx.API, I would suggest asking in the [Checkmarx.API repo](https://github.com/portilha/Checkmarx.API).

Alternatively, it seems that there is a [CLI available](https://checkmarx.atlassian.net/wiki/spaces/SD/pages/1339424904/CLI+Plugin). You can use that using the one of the [process aliases](https://cakebuild.net/api/Cake.Common/ProcessAliases/).

Probably something like:

```cakescript
Task("Scan")
 .Does(() => 
{
  StartProcess("runCxConsole.cmd", new ProcessSettings
  { 
    Arguments = @"Scan -v -ProjectName ""CxServer/bookname j2"" -CxServer http://localhost -CxUser username -CxPassword admin -LocationType folder -LocationPath ""C:\Data\Projects\Java\bs java"" -preset ""Checkmarx Default""" 
  });
});
```

(Note: I took the Arguments to `runCxConsole.cmd` [from the documentation](https://checkmarx.atlassian.net/wiki/spaces/SD/pages/3053847557/Logging+on+to+the+CLI) - I did not test that.)
